runelite-api/src/main/java/net/runelite/api/geometry/Geometry.java

```diff
@@ -0,0 +1,464 @@
+	 * @param y1 Y coordinate of the first endpoint of the line.
+	 * @param x2 X coordinate of the second endpoint of the line.
+	 * @param y2 Y coordinate of the second endpoint of the line.
+	 */
+	private static void appendUnitLines(GeneralPath path, float unitSize,
+		float x1, float y1, float x2, float y2)
+	{
+		float x = x1;
+		float y = y1;
+		float angle = (float)Math.atan2(y2 - y1, x2 - x1);
+		float dx = (float)Math.cos(angle) * unitSize;
+		float dy = (float)Math.sin(angle) * unitSize;
+		float length = (float)Math.hypot(x2 - x1, y2 - y1);
+		int steps = (int)((length - 1e-4) / unitSize);
+		for (int i = 0; i < steps; i++)
+		{
+			x += dx;
+			y += dy;
+			path.lineTo(Math.round(x), Math.round(y));
+		}
+	}
+
+	/**
+	 * Splits a path into smaller segments.
+	 * For example, calling this on a path with a line of length 6, with desired
+	 * unit size of 2, would split the path into 3 consecutive lines of length 2.
+	 *
+	 * @param it The iterator of the path to modify.
+	 * @param unitSize The desired length to use for the segments.
+	 * @return The modified path.
+	 */
+	public static GeneralPath unitifyPath(PathIterator it, float unitSize)
+	{
+		GeneralPath newPath = new GeneralPath();
+		float[] prevCoords = new float[2];
+		float[] coords = new float[2];
+		float[] startCoords = null;
+		while (!it.isDone())
+		{
+			int type = it.currentSegment(coords);
+			if (type == PathIterator.SEG_MOVETO)
+			{
+				if (startCoords == null)
+				{
+					startCoords = new float[2];
+					startCoords[0] = coords[0];
+					startCoords[1] = coords[1];
+				}
+				newPath.moveTo(coords[0], coords[1]);
+				prevCoords[0] = coords[0];
+				prevCoords[1] = coords[1];
+			}
+			else if (type == PathIterator.SEG_LINETO)
+			{
+				appendUnitLines(newPath, unitSize, prevCoords[0], prevCoords[1], coords[0], coords[1]);
+				newPath.lineTo(coords[0], coords[1]);
+				prevCoords[0] = coords[0];
+				prevCoords[1] = coords[1];
+			}
+			else if (type == PathIterator.SEG_CLOSE)
+			{
+				appendUnitLines(newPath, unitSize, coords[0], coords[1], startCoords[0], startCoords[1]);
+				newPath.closePath();
+				startCoords = null;
+			}
+			it.next();
+		}
+
+		return newPath;
+	}
+
+	/**
+	 * Splits a path into smaller segments.
+	 * For example, calling this on a path with a line of length 6, with desired
+	 * unit size of 2, would split the path into 3 consecutive lines of length 2.
+	 *
+	 * @param path The path to modify.
+	 * @param unitSize The desired length to use for the segments.
+	 * @return The modified path.
+	 */
+	public static GeneralPath unitifyPath(GeneralPath path, float unitSize)
+	{
+		return unitifyPath(path.getPathIterator(new AffineTransform()), unitSize);
+	}
+
+	/**
+	 * Removes lines from a path according to a method.
+	 *
+	 * @param it The iterator of the path to filter.
+	 * @param method The method to use to decide which lines to remove. Takes two float[2] arrays with x and y coordinates of the endpoints of the line.
```

Comment:
This doesn't really make it clear whether a `true` or `false` test would indicate a line should be removed.

Reply:
Agreed, added some extra info.